jenkins/automation/kubernetes/k8-cicd-motr.groovy

```diff
@@ -0,0 +1,334 @@
+		
+		// 'WARNING' - rm -rf command used on this path please careful when updating this value
+		
+		DESTINATION_RELEASE_LOCATION = "/mnt/bigstorage/releases/cortx/github/pr-build/${BRANCH}/${COMPONENT_NAME}/${BUILD_NUMBER}"
+		PYTHON_DEPS = "/mnt/bigstorage/releases/cortx/third-party-deps/python-deps/python-packages-2.0.0-latest"
+		COMPONENTS_RPM = "/mnt/bigstorage/releases/cortx/components/github/${BRANCH}/${OS_VERSION}/dev/"
+		CORTX_BUILD = "http://cortx-storage.colo.seagate.com/releases/cortx/github/pr-build/${BRANCH}/${COMPONENT_NAME}/${BUILD_NUMBER}"
+
+		// Artifacts location
+		
+		CORTX_ISO_LOCATION = "${DESTINATION_RELEASE_LOCATION}/cortx_iso"
+		THIRD_PARTY_LOCATION = "${DESTINATION_RELEASE_LOCATION}/3rd_party"
+		PYTHON_LIB_LOCATION = "${DESTINATION_RELEASE_LOCATION}/python_deps"
+		
+		////////////////////////////////// DEPLOYMENT VARS /////////////////////////////////////////////////////
+		// NODE1_HOST - Env variables added in the node configurations
+		
+		build_identifier = sh(script: "echo ${CORTX_BUILD} | rev | cut -d '/' -f2,3 | rev", returnStdout: true).trim()
+		STAGE_DEPLOY = "yes"
+	}
+	stages {
+		stage ("Define Variable") {
+			steps {
+				script { build_stage = env.STAGE_NAME }
+				script {
+					env.allhost = sh( script: '''
+					echo $hosts | tr ' ' '\n' | awk -F["="] '{print $2}'|cut -d',' -f1
+					''', returnStdout: true).trim()
+				}
+			}
+		}
+		// Build motr fromm PR source code
+		stage('Build') {
+			steps {
+				script { build_stage = env.STAGE_NAME }
+				script { manager.addHtmlBadge("&emsp;<b>Target Branch : ${BRANCH}</b>&emsp;<br />") }
+				sh """
+					set +x
+					echo "--------------BUILD PARAMETERS -------------------"
+					echo "MOTR_URL              = ${MOTR_URL}"
+					echo "MOTR_BRANCH           = ${MOTR_BRANCH}"
+					echo "MOTR_PR_REFSEPEC       = ${MOTR_PR_REFSEPEC}"
+					echo "-----------------------------------------------------------"
+				"""
+				dir("motr") {
+					checkout([$class: 'GitSCM', branches: [[name: "${MOTR_BRANCH}"]], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'AuthorInChangelog'], [$class: 'SubmoduleOption', disableSubmodules: false, parentCredentials: true, recursiveSubmodules: true, reference: '', trackingSubmodules: false]], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: "${MOTR_URL}",  name: 'origin', refspec: "${MOTR_PR_REFSEPEC}"]]])
+					sh label: '', script: '''
+						yum install kernel-devel -y
+						export build_number=${BUILD_ID}
+						kernel_src=$(ls -1rd /lib/modules/*/build | head -n1)
+						cp cortx-motr.spec.in cortx-motr.spec
+						sed -i 's/@.*@/111/g' cortx-motr.spec
+						yum-builddep -y cortx-motr.spec
+					'''
+					
+					sh label: '', script: '''
+						rm -rf /root/rpmbuild/RPMS/x86_64/*.rpm
+						KERNEL=/lib/modules/$(yum list installed kernel | tail -n1 | awk '{ print $2 }').x86_64/build
+						./autogen.sh
+						./configure --with-linux=$KERNEL
```

Comment:
Please use user-mode compilation for K8s pre-merge. Please refer - https://github.com/Seagate/cortx-re/blob/kubernetes/jenkins/internal-ci/centos-7.9.2009/branch/motr.groovy#L89

Reply:
Replace below line.

./configure --with-linux=$KERNEL to ./configure --with-user-mode-only